src/core/session.js

```diff
@@ -91,26 +116,54 @@ export class Session {
     }
   }
 
+  /**
+   * Connects a new stream source.
+   *
+   * @param {any} source - The new stream source to connect.
```

Comment:
This is either [EventSource](https://developer.mozilla.org/en-US/docs/Web/API/EventSource) | [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket)